test/active_reporting/report_test.rb

```diff
@@ -32,8 +32,23 @@ def test_report_runs_with_an_aggregate_other_than_count
     assert data.all? { |r| r.key?('a_metric') }
   end
 
-  def test_report_runs_with_a_date_grouping
-    if ENV['DB'] == 'pg'
+  def test_report_runs_with_week_date_grouping
+    if valid_db_adapter?
+      metric = ActiveReporting::Metric.new(:a_metric, fact_model: SaleFactModel, dimensions: [{created_at: :week}])
+      report = ActiveReporting::Report.new(metric)
+      data = report.run
+      assert data.all? { |r| r.key?('created_at_week') }
+      assert data.size == 4
+    else
+      assert_raises ActiveReporting::InvalidDimensionLabel do
+        metric = ActiveReporting::Metric.new(:a_metric, fact_model: SaleFactModel, dimensions: [{created_at: :week}])
+        report = ActiveReporting::Report.new(metric)
+      end
+    end
+  end
```

Comment:
I would propose that you might want to add some more comprehensive tests for the MySQL date_trunc adaptation. You can run into **a lot** of edge cases. See [here](https://github.com/LawKick/active_reporting/blob/master/test/active_reporting/function_adapaters/date_trunc_test.rb).